src/molinit.c

```diff
@@ -187,14 +188,28 @@ molinit(molData *m, inputPars *par, struct grid *g,int i){
     part = malloc(sizeof(struct data) * m[i].npart);
 
     for(ipart=0;ipart<m[i].npart;ipart++){
-      fgets(string, 80, fp);
-      fscanf(fp,"%d\n", &count[ipart]);
-      fgets(string, 80, fp);
-      fgets(string, 80, fp);
+      fgets(string, sizeI, fp);
+      fscanf(fp,"%d\n", &collPartIDs[ipart]);
+
+      /* We want to test if the comment after the coll partner ID number is longer than the buffer size. To do this, we write a character - any character, as long as it is not \0 - to the last element of the buffer:
```

Comment:
No. The intent of this elaborate check is purely to avoid a cock-up if the comment line exceeds the buffer length. It is essentially an error trap.